### PR TITLE
[WIN32SS] Fix IntSynthesizeDib to synthesize CF_DIB with CF_BITMAP

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
This PR will fix the IntSynthesizeDib function to synthesize CF_DIB with CF_BITMAP clipboard format.

JIRA issue: [CORE-14770](https://jira.reactos.org/browse/CORE-14770)

### CODE REVIEW

**Comment by katahiromz** [comment]
AFTER:
dib-after

File: win32ss/user/ntuser/clipboard.c [hunk 1]
```diff
@@ -160,7 +161,16 @@ IntSynthesizeDib(
...
     /* Get information about the bitmap format */
+    if (!GreGetObject(hbm, sizeof(BITMAP), &bm))
+        goto cleanup;
+
+    memset(&bmiBuffer, 0, sizeof(bmiBuffer));
     pbmi->bmiHeader.biSize = sizeof(bmiBuffer.bmih);
+    pbmi->bmiHeader.biWidth = bm.bmWidth;
+    pbmi->bmiHeader.biHeight = bm.bmHeight;
+    pbmi->bmiHeader.biPlanes = bm.bmPlanes;
+    pbmi->bmiHeader.biBitCount = bm.bmBitsPixel;
```
Comment (HBelusca) [inline win32ss/user/ntuser/clipboard.c]: I would have expected that it's done inside GreGetDIBitsInternal() code, as it seems to be already done in lines 769 and beyond in [here](https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/gdi/ntgdi/dibobj.c;hb=4d494caf614b03d8ecec09deaf53c5fe96f01e73#l669) ?

Comment (katahiromz) [reply win32ss/user/ntuser/clipboard.c]: The DIB_GetBitmapInfo call in GreGetDIBitsInternal needs BITMAPINFO initialization.

Comment (HBelusca) [reply win32ss/user/ntuser/clipboard.c]: Yes, but not in this case: Normally because GreGetDIBitsInternal is called with its Bits parameter == NULL, by using the pbmi parameter initialized with just these two lines:
```c
     memset(&bmiBuffer, 0, sizeof(bmiBuffer));
     pbmi->bmiHeader.biSize = sizeof(bmiBuffer.bmih);
```
it should call DIB_GetBitmapInfo() that must return bitmap_type  == 1 while all the width, height, etc... and bpp parameters set to zero. The following "Validate input" checks should all pass, and the
```c
 769     /* Fill in the structure */
 770     switch(bpp)
 771     {
 772     case 0: /* Only info */
```
case should run, and in return one should obtain correct values in pbmi->bmiHeader.

In other words, this very first GreGetDIBitsInternal() call should do what GreGetObject() does.

Can you check in more details what exactly fails in that first GreGetDIBitsInternal() call: are the validation calls correctly run?, is the `case 0: /* Only info */` block executed ?

Comment (katahiromz) [reply win32ss/user/ntuser/clipboard.c]: No, GreGetDIBitsInternal is not GetDIBits.

Comment (HBelusca) [reply win32ss/user/ntuser/clipboard.c]: yes, bpp will be zero, but then aren't the Info->bmpStuff that are initialized in the `case 0: /* Only info */` case correctly obtained/calculated then?

Comment (katahiromz) [reply win32ss/user/ntuser/clipboard.c]: There is no "memset(&bmiBuffer, 0, sizeof(bmiBuffer));" statement in current codes.
Assume pbmi->bmiHeader.biSize == sizeof(bmiBuffer.bmih) == sizeof(BITMAPINFOHEADER).
so bitmap_type will be 1. And then bpp will be a random value (from uninitialized Info.bmiHeader.biBitCount) and not zero.

Comment (HBelusca) [reply win32ss/user/ntuser/clipboard.c]: Yes yes, that's why I proposed
> by using the pbmi parameter initialized *with just these two lines*:
>     memset(&bmiBuffer, 0, sizeof(bmiBuffer));
>     pbmi->bmiHeader.biSize = sizeof(bmiBuffer.bmih);
(and nothing else)

Comment (katahiromz) [reply win32ss/user/ntuser/clipboard.c]: OK. I will fix.

Comment (HBelusca) [reply win32ss/user/ntuser/clipboard.c]: > OK. I will fix.
Let me know whether it actually works :D

Comment (katahiromz) [reply win32ss/user/ntuser/clipboard.c]: It works. Just one line.
it-works
